Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteGPUProxy.h

```diff
@@ -94,22 +88,19 @@ class RemoteGPUProxy final : public PAL::WebGPU::GPU, private IPC::Connection::C
     {
         return root().streamClientConnection().sendSync(WTFMove(message), backing(), defaultSendTimeout);
     }
-    IPC::Connection& connection() const { return m_gpuProcessConnection->connection(); }
+    IPC::Connection* connection() const { return m_remoteRenderingBackendProxy.connection(); }
 
     void requestAdapter(const PAL::WebGPU::RequestAdapterOptions&, CompletionHandler<void(RefPtr<PAL::WebGPU::Adapter>&&)>&&) final;
 
     Ref<PAL::WebGPU::PresentationContext> createPresentationContext(const PAL::WebGPU::PresentationContextDescriptor&) final;
 
     Ref<PAL::WebGPU::CompositorIntegration> createCompositorIntegration() final;
 
-    void abandonGPUProcess();
-    void disconnectGpuProcessIfNeeded();
-
     Deque<CompletionHandler<void(RefPtr<PAL::WebGPU::Adapter>&&)>> m_callbacks;
 
     WebGPUIdentifier m_backing;
     Ref<WebGPU::ConvertToBackingContext> m_convertToBackingContext;
-    GPUProcessConnection* m_gpuProcessConnection;
+    RemoteRenderingBackendProxy& m_remoteRenderingBackendProxy;
```

Comment:
How do we guarantee that we always destroy all RemoteGPUProxy instances before the RemoteRenderingBackendProxy?

I *think* we should, but RemoteGPUProxy is refcounted (and RemoteRenderingBackendProxy is not), so it's hard to be confident (and confident that it won't change quietly in the future).

I think it's worth making this a WeakPtr (or at least a CheckedPtr) to avoid this.

Reply:
Kimmo has a solution for this in https://github.com/WebKit/WebKit/pull/12765#discussion_r1168554414 - let me see if I can use it here.